src/transformers/commands/serving.py

```diff
@@ -12,32 +12,55 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import json
+import time
 from argparse import ArgumentParser, Namespace
-from typing import Any, List, Optional
+from dataclasses import dataclass, field
+from logging import getLevelNamesMapping
+from threading import Thread
+from typing import Any, Dict, Optional
 
-from ..pipelines import Pipeline, get_supported_tasks, pipeline
-from ..utils import logging
+from transformers.utils.import_utils import is_fastapi_available, is_pydantic_available, is_uvicorn_available
+
+from .. import PreTrainedTokenizerFast, TextIteratorStreamer
+from ..generation.continuous_batching import ContinuousBatchingManager, RequestStatus
+from ..utils import is_torch_available, logging
 from . import BaseTransformersCLICommand
 
 
-try:
-    from fastapi import Body, FastAPI, HTTPException
-    from fastapi.routing import APIRoute
-    from pydantic import BaseModel
-    from starlette.responses import JSONResponse
-    from uvicorn import run
+if is_torch_available():
+    import torch
 
-    _serve_dependencies_installed = True
-except (ImportError, AttributeError):
-    BaseModel = object
+    from transformers import (
+        AutoModelForCausalLM,
+        AutoTokenizer,
+        BitsAndBytesConfig,
+        GenerationConfig,
+        PreTrainedModel,
+    )
 
-    def Body(*x, **y):
-        pass
 
-    _serve_dependencies_installed = False
+if is_pydantic_available():
+    import uvicorn
+    from fastapi import FastAPI
+    from fastapi.responses import StreamingResponse
+    from pydantic import BaseModel
+
+    class Message(BaseModel):
+        role: str
+        content: str
 
+    class ChatCompletionInput(BaseModel):
+        messages: list[Message]
+        stream: Optional[bool] = False
+        model: Optional[str] = None
+        generation_config: Optional[dict] = None
+        request_id: Optional[str] = None
+        extra_body: Optional[Dict] = None
+        tools: Any = None
```

Comment:
For reference, this is the definition we aim for as `ChatCompletionInput`: https://github.com/huggingface/huggingface_hub/blob/28554021b7655a7d844c32e423d18a5ee1845599/src/huggingface_hub/inference/_generated/types/chat_completion.py#L125. Sad but it's not a pydantic base model so you won't be able to use it directly. There are some threads on google/stackoverflow/github over how to convert a dataclass to a pydantic base model but I've never tried them myself. 

So for now I don't have much suggestion, just wanted to flag the target data structure (even though not everything has to be implemented)